api/v1/stocks.go

```diff
@@ -29,24 +30,46 @@ func (api *APIHandler) getAllStocks(w http.ResponseWriter, r *http.Request) {
 
 func (api *APIHandler) getStockDetails(w http.ResponseWriter, r *http.Request) {
 	symbol := chi.URLParam(r, "symbol")
+	time := chi.URLParam(r, "time")
 
 	yamlData, err := details.StocksYamlBytes()
 	if err != nil {
 		responses.ErrorResponse(w, http.StatusInternalServerError, fetchingError.Error())
+		return
 	}
 
 	var myStocks []details.StockDetails
-
 	if err := yaml.Unmarshal(yamlData, &myStocks); err != nil {
-		responses.ErrorResponse(w, http.StatusInternalServerError, fetchingError.Error())
+		responses.ErrorResponse(w, http.StatusInternalServerError, decodingError.Error())
+		return
 	}
 
-	for i := range myStocks {
-		if myStocks[i].Symbol == symbol {
-			responses.CustomResponse(w, myStocks[i], 200)
+	stock, err := getStockHelper(myStocks, symbol)
+	if err != nil {
+		responses.ErrorResponse(w, http.StatusInternalServerError, stockNotFoundError.Error())
+		return
+	}
+	if time != "" {
+		tick := timeutils.GetTickAtTime(time)
+		stockMapAtTick, err := api.DB.GetStockMapAtTick(tick)
```

Comment:
We should also consider adding a database function to only get a single stock for a specified tick

Reply:
@explodingcamera I changed all the things you mentioned except for this one, since I thought that might be something either Alex, Andre or you might be interested in doing (or more adept in doing this since you and Alex worked on the database), and then I could change this portion later on too as soon as the respective function exists :0 Let me know what you think and thanks again for all the feedback and constructive criticism! ( ͡°⁄ ⁄ ͜⁄ ⁄ʖ⁄ ⁄ ͡°)

Reply:
Awesome, yeah this was more meant as a general comment, I'll open a new github issue.